drivers/sensor/bmi08x/bmi08x_accel.c

```diff
@@ -0,0 +1,775 @@
+ */
+#define BMI08X_DATA_SYNC_REG_VAL(inst)                                                             \
+	(BMI08X_GYRO_ODR(inst) == 3 && BMI08X_ACCEL_ODR(inst) == 5)   ? 3                          \
+	: (BMI08X_GYRO_ODR(inst) == 2 && BMI08X_ACCEL_ODR(inst) == 6) ? 2                          \
+	: ((BMI08X_GYRO_ODR(inst) == 1 || BMI08X_GYRO_ODR(inst) == 0) &&                           \
+	   BMI08X_ACCEL_ODR(inst) == 7)                                                            \
+		? 1                                                                                \
+		: 0
+
+/* define the .data_sync in the driver config */
+#if BMI08X_ACCEL_ANY_INST_HAS_DATA_SYNC
+/* if another bmi08x as the data sync enabled, and one doesn't, it will get the value of 0 and won't
+ * have the config file sent over
+ */
+#define BMI08X_DATA_SYNC_REG(inst)                                                                 \
+	.data_sync = COND_CODE_1(BMI08X_ACCEL_DATA_SYNC_EN(inst),                                  \
+				 (BMI08X_DATA_SYNC_REG_VAL(inst)), (0)),
+#define BMI08X_ACCEL_TRIGGER_PINS(inst) BMI08X_ACCEL_TRIG(inst)
+#else
+#define BMI08X_DATA_SYNC_REG(inst)
+#define BMI08X_ACCEL_TRIGGER_PINS(inst)                                                            \
+	IF_ENABLED(CONFIG_BMI08X_ACCEL_TRIGGER, (BMI08X_ACCEL_TRIG(inst)))
+#endif
+
+#define BMI08X_CREATE_INST(inst)                                                                   \
+                                                                                                   \
+	IF_ENABLED(BMI08X_ACCEL_DATA_SYNC_EN(inst), (BMI08X_VERIFY_DATA_SYNC(inst);))              \
+	IF_ENABLED(BMI08X_ACCEL_DATA_SYNC_EN(inst), (BMI08X_VERIFY_DATA_SYNC_ODR(inst);))          \
+	IF_ENABLED(BMI08X_ACCEL_DATA_SYNC_EN(inst), (BMI08X_VERIFY_GYRO_DATA_SYNC_EN(inst);))      \
+                                                                                                   \
+	static struct bmi08x_accel_data bmi08x_drv_##inst;                                         \
+                                                                                                   \
+	static const struct bmi08x_accel_config bmi08x_config_##inst = {                           \
+		COND_CODE_1(DT_INST_ON_BUS(inst, spi), (BMI08X_CONFIG_SPI(inst)),                  \
+			    (BMI08X_CONFIG_I2C(inst)))                                             \
+			.api = COND_CODE_1(DT_INST_ON_BUS(inst, spi), (&bmi08x_spi_api),           \
+					   (&bmi08x_i2c_api)),                                     \
+		IF_ENABLED(CONFIG_BMI08X_ACCEL_TRIGGER,                                            \
+			   (.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),))                 \
+			BMI08X_ACCEL_TRIGGER_PINS(inst)                                            \
+				.accel_hz = DT_INST_ENUM_IDX(inst, accel_hz) + 5,                  \
+		.accel_fs = DT_INST_ENUM_IDX(inst, accel_fs) / 2, BMI08X_DATA_SYNC_REG(inst)};     \
```

Comment:
.accel_fs = DT_INST_ENUM_IDX(inst, accel_fs) / 2

this is a bug, you need change to DT_INST_PROP(inst, accel_fs)
because in `bmi08x_acc_range_set(dev, config->accel_fs);` the range parameter need the real rang not the index. 

same as gyro sensor fs setting

Reply:
Nope, the enum idx is to correspond with the reg value, please take a look at the datasheet again, the divide by 2 is there because the bmi085 and bmi088 have different settings

Reply:
> 
first, I have tested this is mistake. I tried to explain why it was a mistake
In function `static int bmi08x_acc_range_set(const struct device *dev, int32_t range)`, called in `bmi08x_accel_init` to setting accel sensor range, the parameter range is using to get register value by `reg_val = bmi08x_range_to_reg_val(range, bmi085_acc_range_map, BMI085_ACC_RANGE_MAP_SIZE);`

and in function `bmi08x_range_to_reg_val`
```
int32_t bmi08x_range_to_reg_val(uint16_t range, const struct bmi08x_range *range_map,
				       uint16_t range_map_size)
{
	int i;

	for (i = 0; i < range_map_size; i++) {
		if (range <= range_map[i].range) {
			return range_map[i].reg_val;
		}
	}

	return -EINVAL;
}```

the range parameter is compare with 
```
static const struct bmi08x_range bmi085_acc_range_map[] = {
	{2, BMI085_ACCEL_RANGE_2G},
	{4, BMI085_ACCEL_RANGE_4G},
	{8, BMI085_ACCEL_RANGE_8G},
	{16, BMI085_ACCEL_RANGE_16G},
};
```
you can see that range map value is `2, 4, 8, 16`, this is range value not index.

Reply:
I see, sorry, I was confused with another definition in my head. I will fix

Reply:
> I see, sorry, I was confused with another definition in my head. I will fix

same as gyro sensor range setting, and `static const struct bmi08x_range bmi08x_gyr_range_map[] = {` need sort from small to large.

Reply:
> > I see, sorry, I was confused with another definition in my head. I will fix
> 
> same as gyro sensor range setting, and `static const struct bmi08x_range bmi08x_gyr_range_map[] = {` need sort from small to large.

done